src/snowflake/snowpark/dataframe.py

```diff
@@ -916,9 +916,17 @@ def to_pandas(
             2. If you use :func:`Session.sql` with this method, the input query of
             :func:`Session.sql` can only be a SELECT statement.
         """
-
         if _emit_ast:
```

Comment:
the stmt code here looks good to me! I think what is missing is an eval. `to_pandas` should be similar to collect() that it should issue the REST request. Cf. the implementation of `collect()` for example. If you write a test like

```
df = ...
with session.ast_listener() as al:
   df.to_pandas()
assert al.base64_batches is not None # <-- should contain stmt + eval. Flushed via `dataframeAst` property as part of json.
```

Reply:
Added eval